warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -5208,74 +5213,78 @@ public static void reset(GeoTimeSerie gts) {
       long start = tick;
       long stop = tick;
 
-      if (prewindow < 0) {
-        start = tick + prewindow;
-      } else if (prewindow > 0) {
-        // window is a number of ticks
-        if (null == ticks) {
-          if (null == outputTicks || !reversed) {
-            start = prewindow <= mapped.bucketcount ? tick - prewindow * mapped.bucketspan : Long.MIN_VALUE;
-          } else {
-            start = prewindow - 1 <= mapped.bucketcount ? tick - (prewindow - 1) * mapped.bucketspan : Long.MIN_VALUE;
-          }
-        } else {
-          if (null == outputTicks) {
-            if (reversed) {
-              start = idx + prewindow < ticks.length ? (ticks[idx + (int) prewindow]) : Long.MIN_VALUE;
+      // we don't need to extract more than a single point if we use a FillerMapper on a bucketized GTS on a non-empty bucket
```

Comment:
Rewrite the comment, the fact that the GTS is bucketized is not directly visible in the condition you inserted. Think about clarity for people reading the comment.